models/login_source.go

```diff
@@ -291,22 +297,53 @@ func GetLoginSourceByID(id int64) (*LoginSource, error) {
 	return source, nil
 }
 
+// ResetNonDefaultLoginSources clean other default source flag
+func ResetNonDefaultLoginSources(source *LoginSource) error {
+	// update changes to DB
+	if _, err := x.NotIn("id", []int64{source.ID}).Cols("is_default").Update(&LoginSource{IsDefault: false}); err != nil {
+		return err
+	}
+	// write changes to local authentications
+	for i := range localLoginSources.sources {
+		localSource := &LoginSource{}
+		*localSource = *localLoginSources.sources[i]
```

Comment:
This for loop only performs read-only operations, so I think we don't need this line here, just directly use `localLoginSources.sources[i]` should be fine.